src/torque-offset-estimator.cpp

```diff
@@ -0,0 +1,265 @@
+                              dynamicgraph::Vector, m_collectSensorDataSINNER)
+        ,sensor_offset_status(PRECOMPUTATION)
+      {
+        Entity::signalRegistration( ALL_INPUT_SIGNALS << ALL_OUTPUT_SIGNALS);
+        addCommand("init", makeCommandVoid2(*this, &TorqueOffsetEstimator::init,
+                                            docCommandVoid2("Initialize the estimator" ,
+                                                            "urdfFilePath",
+                                                            "Definition of the imu in the chest frame")));
+        addCommand("computeOffset",
+                   makeCommandVoid2(*this, &TorqueOffsetEstimator::computeOffset,
+                                    docCommandVoid2("Compute the offset for sensor calibration",
+                                                    "Number of iteration to average over.",
+                                                    "Maximum allowed offset")));
+        addCommand("getSensorOffsets",
+                   makeDirectGetter(*this,&jointTorqueOffsets,
+                                    docDirectGetter("Return the computed sensor offsets",
+                                                    "vector")));
+
+        encSignals.clear();
+        accSignals.clear();
+        gyrSignals.clear();
+        tauSignals.clear();
+        
+      }
+
+
+      /* --- COMMANDS ---------------------------------------------------------- */
+      /* --- COMMANDS ---------------------------------------------------------- */
+      /* --- COMMANDS ---------------------------------------------------------- */
+      void TorqueOffsetEstimator::init(const std::string &urdfFile,
+                                       const Eigen::Matrix4d& _m_torso_X_imu)
+      {
+        try {
+          se3::urdf::buildModel(urdfFile,se3::JointModelFreeFlyer(),m_model);
+          assert(m_model.nq == N_JOINTS+7);
+          assert(m_model.nv == N_JOINTS+6);
+          jointTorqueOffsets.resize(N_JOINTS);
+          jointTorqueOffsets.setZero();
+        }
+        catch (const std::exception& e) 
+        { 
+          std::cout << e.what();
+          return SEND_MSG("Init failed: Could load URDF :" + urdfFile, MSG_TYPE_ERROR);
+        }
+        m_data = new se3::Data(m_model);
+        m_torso_X_imu.rotation() = _m_torso_X_imu.block<3,3>(0,0);
+        m_torso_X_imu.translation() = _m_torso_X_imu.block<3,1>(0,3);
+      }
+
+      void TorqueOffsetEstimator::computeOffset(const int& nIterations, const double& _epsilon)
+      {
+        if (sensor_offset_status == PRECOMPUTATION) {
+          SEND_MSG("Starting offset computation with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+          n_iterations = nIterations;
+          sensor_offset_status = INPROGRESS;
+        }
+        else if (sensor_offset_status == INPROGRESS) {
+          SEND_MSG("Collecting input signals. Please keep the graph running", MSG_TYPE_WARNING_STREAM);
```

Comment:
You don't wanna use MSG_TYPE_WARNING_STREAM here because it's not inside the control loop but it's inside a command, so it's gonna be called only once. Replace it with MSG_TYPE_WARNING.